src/util/event_loop_dispatcher.hpp

```diff
@@ -85,6 +85,25 @@ class EventLoopDispatcher<void(Args...)> {
         m_scheduler->notify();
     }
 };
+
+namespace detail {
+template <typename T>
+struct extract_signature_impl {};
+template <typename Sig>
+struct extract_signature_impl<std::function<Sig>> {
+    using signature = Sig;
+};
+
+template <typename T>
+using extract_signature = typename extract_signature_impl<T>::signature;
+}
+
+// Use std::function deduction guides for EventLoopDispatcher. This works with function pointers, lambdas (without
```

Comment:
I could write out a version of `extract_signature` that doesn't rely on `std::function` if you'd prefer. It isn't complicated, just long(ish) and repetitive. And since we are already using a `std::function` as the constructor argument, it even seems vaguely correct to just defer to std::function, rather than defining our own.

Reply:
If you are curious, this is the libc++ `std::function` deduction guide impl: https://github.com/llvm/llvm-project/blob/fb38b98338cc87442e3451665e82bf1c8ef9388f/libcxx/include/functional#L2428-L2473

Reply:
Blarg. Looks like libc++ added the `std::function` deduction guides too recently for them to be available in the version of xcode we are using: https://github.com/llvm/llvm-project/commit/e1eabcdfad89f67ae575b0c86aa4a72d277378b4. I guess I'll have to not rely on them here anyway.

I'll do that tomorrow. No need to review until then, unless you think you are opposed to the general idea of this commit.